### ISSUE TITLE
Error validating transaction: witness script detected in tx without witness data.

### ISSUE THREAD

**Issue by maksus** [Bit-Wasp/bitcoin-php#807]
Hello!
Cannot figure it out for 37 sleepless hours.
 May be misunderstood the signer logic or Blockchain specific..
Tried to make the code as simple as possible.

         ` $privKeyFactory = new PrivateKeyFactory();
           $privateKey = $privKeyFactory->fromWif("p2pkh:PRIVATE_KEY");
          
             $program = ScriptFactory::scriptPubKey()->payToPubKeyHash($privateKey->getPubKeyHash());
                $txOut = new TransactionOutput(100000, $program);
    
      $addressCreator = new AddressCreator();
     $transaction = TransactionFactory::build()
    ->input('76a44ad2ed45c37a03fb5ca77f6a842152d85a46fbe6d371324742011031968a', 0)
    ->payToAddress(90000, $addressCreator->fromString('181ahqXXRpEyQ7Qsav7p7DrWF7t19LTSpv'))
    ->get();
      
       $signer = new Signer($transaction);
      $signer -> sign (0, $privateKey, $txOut);
      $signed = $signer->get();

 `

PS The library is super-cool and extremely useful.. Thanks!

**Comment by afk11** [comment]
Hmm, not aware of an issue like that right now. I'll take a little look tonight!

**Comment by maksus** [comment]
Yes - this is TEST ADDRESS. 
You saved me from going crazy about this problem... 
So this number in "input('hash', 'NUMBER')" means the id of the UTXO? 

Thanks once again.

**Comment by afk11** [comment]
> So this number in "input('hash', 'NUMBER')" means the id of the UTXO?

Yes - it's the id of the output of transaction identified by 'hash'. And when you're signing, you provide the full transaction output (value, and scriptPubKey) for $hash, $id

Sure, I got this one! bc1qp8ys9s3tvrjdv40m26tg0r2rlwtg6yeadqu56z

Cheers!

**Comment by Lutnerd** [comment]
Good this transaction is giving me this ,Error validating transaction: witness script detected in tx without witness data.
